Resources/Prototypes/Entities/Structures/Power/Generation/generators.yml

```diff
@@ -0,0 +1,265 @@
+  - type: Sprite
+    drawdepth: WallMountedItems
+    netsync: false
+    sprite: Structures/Power/Generation/wallmount_generator.rsi
+    layers:
+    - state: panel
+  - type: Construction
+    graph: WallmountGenerator
+    node: frame
+
+# Generators in use
+
+- type: entity
+  parent: BaseGenerator
+  id: GeneratorBasic
+  suffix: Basic, 3kW
+  components:
+  - type: PowerSupplier
+    supplyRate: 3000
+    
+- type: entity
+  parent: BaseGenerator
+  id: GeneratorPlasma
+  suffix: Plasma, 5kW
+  components:
+  - type: PowerSupplier
+    supplyRate: 5000 # 260 sec / sheet
+  - type: Sprite
+    sprite: Structures/Power/Generation/portable_generator.rsi
+    state: portgen0_1
+  - type: Construction
+    graph: Machine
+    node: machine
+  - type: Wires
+    BoardName: "GeneratorPlasma"
+    LayoutId: GeneratorPlasma
+  - type: Machine
+    board: GeneratorPlasmaMachineCircuitboard
+    
+- type: entity
+  parent: BaseGenerator
+  id: GeneratorUranium
+  suffix: Uranium, 15kW
+  components:
+  - type: PowerSupplier
+    supplyRate: 15000 # 85 sec / sheet
+  - type: Sprite
+    sprite: Structures/Power/Generation/portable_generator.rsi
+    state: portgen1_1
+  - type: Construction
+    graph: Machine
+    node: machine
+  - type: Wires
+    BoardName: "GeneratorUranium"
+    LayoutId: GeneratorUranium
+  - type: Machine
+    board: GeneratorUraniumMachineCircuitboard
+
+
+- type: entity
+  parent: BaseGeneratorWallmount
+  id: GeneratorWallmountBasic
+  suffix: Basic, 3kW
+  components:
+  - type: PowerSupplier
+    supplyRate: 3000
+  - type: Construction
+    graph: WallmountGenerator
+    node: generator
+
+- type: entity
+  parent: BaseGeneratorWallmount
+  id: GeneratorWallmountAPU
+  name: shuttle APU
+  description: An auxiliary power unit for a shuttle - 6kW.
+  suffix: APU, 6kW
+  components:
+  - type: PowerSupplier
+    supplyRate: 6000
+  - type: Construction
+    graph: WallmountGenerator
+    node: APU  
+
+# RTG - no fuel requirement
+
+- type: entity
+  parent: BaseGenerator
+  id: GeneratorRTG
+  name: RTG
+  description: A Radioisotope Thermoelectric Generator for long term power.
+  suffix: 20kW
+  components:
+  - type: PowerSupplier
+    supplyRate: 20000 # supposed to be low power at no fuel cost to power derelicts
+  - type: Sprite
+    sprite: Structures/Power/Generation/rtg.rsi
+    state: rtg
+# TODO: need radioactive fallout when destroyed
+
+- type: entity
+  parent: BaseGenerator
+  id: GeneratorRTGDamaged
+  name: damaged RTG
+  description: A Radioisotope Thermoelectric Generator for long term power. This one has damaged shielding.
+  suffix: 20kW
+  components:
+  - type: PowerSupplier
+    supplyRate: 20000
+  - type: Sprite
+    netsync: false
+    sprite: Structures/Power/Generation/rtg.rsi
+    layers:
+      - state: rtg_damaged
+      - state: rtg_glow
+  - type: RadiationPulse # ideally only when opened.
+    range: 2
+    decay: false
+    minPulseLifespan: 1
+    maxPulseLifespan: 1
```

Comment:
Bit suss having this as there's no way to detect radiation atm; at least with singulo and radstorm it's obvious.

Reply:
The RTG has the glowing overlay and since it's only 1 tile distance you'll sharp notice you're taking damage if you're pulling it not wearing something. Darwin Award at that point going at a leaking nuclear source without a suit.